Tasks/JavaToolInstallerV0/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -142,55 +144,80 @@ export class JavaFilesExtractor {
      * @param pathsArray - contains paths to all the files inside the structure
      * @param root - path to the directory we want to get the structure of
      */
-    private static sliceStructure(pathsArray: Array<string>, root: string = pathsArray[0]): Array<string>{
+    public static sliceStructure(pathsArray: Array<string>, root: string = pathsArray[0]): Array<string>{
         const dirPathLength = root.length;
         const structureObject: IDirectoriesDictionary = {};
         for(let i = 0; i < pathsArray.length; i++){
             const pathStr = pathsArray[i];
             const cleanPathStr = pathStr.slice(dirPathLength + 1);
+            if (cleanPathStr === '') continue;
             const dirPathArray = cleanPathStr.split(path.sep);
             // Create the list of unique values
             structureObject[dirPathArray[0]] = null;
         }
         return Object.keys(structureObject);
     }
 
+    /**
+     * Returns name w/o extension
```

Comment:
'file ending' instead of 'extension'? I think we don't need to mix it up - e.g. for file 'file.tar.gz' file extension is .gz, but fileEnding (term used in code) is .tar.gz.